### ISSUE TITLE
What's the problem in providing a up-to-date VM with sudo access?

### ISSUE THREAD

**Issue by Paebbels** [travis-ci/travis-ci#6542]
Ubuntu 16.04 was announced months ago. But now it's scheduled to a *maybe* in the next year, without any statement why and what's the problem ...

Using Ubuntu 14.04 with sudo rights requires many computing resources, because each VM run needs
 * high traffic to download updates and missing packages,
 * high CPU usage to install and upgrade the system,
 * high disk I/O to write the download and to install the new packages (i don't assume that all VMs run purely in main memory ...)
 * people need to maintain the update process

So all in all this costs **you** computation power...
While old 14.04 is doing updates, a new 16.04 could have run build tasks in the meanwhile.

So please explain to your users - who like this free service - what is the problem with Ubuntu 16.04?

Kind regards
    Patrick

**Comment by BanzaiMan** [comment]
I'm closing this as a duplicate of #5821.

We understand the general desire to run on 16.04. There are many things that go into offering that to our users. It is in our long-term interest to make this sort of updates easier and quicker. While we are making progress towards that goal, the availability of 16.04 remains a bit uncertain at this time.

**Comment by joshk** [comment]
Hi @Paebbels 

Thanks for the message.

My name is Josh, I'm the Product Manager here at Travis CI.

Although we have plans for 16.04 support next year, this is all dependent on getting 14.04 stable and shipped to all setups (container based setup, Google Compute, and Enterprise), as well as the other work required for scaling our Mac setup, improving how we run everyones builds (we run roughly 150k jobs per day), and making continuous speed and stability improvements.

What I really mean to say is that we have a lot of work on our backlog, and as much as we want to support newer operating systems, the work required is very significant. For example, updating all Chef recipes, testing, more updating, more testing, and then providing regular updates.

As a minor note, the adoption of 16.04 is not as extensive as 14.04 for production systems, thus testing on 14.04 makes more sense for many people.

The points you make about high traffic, cpu and disk I/O are fairly minimal really in terms of cost.

As a side question, where did we announce 16.04 support? 

Thanks for the great feedback.

Josh

**Comment by Paebbels** [comment]
Hi @joshk,

thank's for the explanations. So it's more than a simple 14.04 replacement :)
So I can understand that it's taking more time.

I found it somewhere on the Internet, that Travis is going to switch 14.04 to 16.04 in mid 2016. Sorry I can't deliver the exact URL. I was just wondering that mid 2016 is over and 16.04 is not available. So others, who might seek for an answer on Ubuntu 16.04, can refer to our Q&A pair :)

Actually, I found that Travis improved the apt installation speed (old runs required circa 50 sec, new runs can do the same update procedure in 20 sec).

Kind regards
    Patrick Lehmann
